pkg/assets/create/creater.go

```diff
@@ -178,36 +178,37 @@ func create(ctx context.Context, manifests map[string]*unstructured.Unstructured
 		}
 		resourceString := mappings.Resource.Resource + "." + mappings.Resource.Version + "." + mappings.Resource.Group + "/" + manifests[path].GetName() + " -n " + manifests[path].GetNamespace()
 
-		incluster, err := resource.Create(ctx, manifests[path], metav1.CreateOptions{})
-
-		if err == nil && options.Verbose {
-			fmt.Fprintf(options.StdErr, "Created %q %s\n", path, resourceString)
-		}
-
-		// Resource already exists means we already succeeded
-		// This should never happen as we remove already created items from the manifest list, unless the resource existed beforehand.
-		if kerrors.IsAlreadyExists(err) {
+		incluster, err := resource.Get(ctx, manifests[path].GetName(), metav1.GetOptions{})
+		if err == nil {
 			if options.Verbose {
 				fmt.Fprintf(options.StdErr, "Skipped %q %s as it already exists\n", path, resourceString)
 			}
-			incluster, err = resource.Get(ctx, manifests[path].GetName(), metav1.GetOptions{})
-			if err != nil {
+			// fall through as if it was just created
+		} else if !kerrors.IsNotFound(err) {
+			if options.Verbose {
+				fmt.Fprintf(options.StdErr, "Failed to get already existing %q %s: %v\n", path, resourceString, err)
```

Comment:
this is now wrong, isn't it? the error is not NotFound, so we don't know whether the resource is existing or not.

Reply:
yes. Copied it from the create case without reading the message :D

Reply:
fixed